src/deluge/gui/views/performance_session_view.cpp

```diff
@@ -0,0 +1,1931 @@
+/*
+ * Copyright © 2014-2023 Synthstrom Audible Limited
+ *
+ * This file is part of The Synthstrom Audible Deluge Firmware.
+ *
+ * The Synthstrom Audible Deluge Firmware is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "gui/views/performance_session_view.h"
+#include "definitions_cxx.hpp"
+#include "dsp/master_compressor/master_compressor.h"
+#include "extern.h"
+#include "gui/colour.h"
+#include "gui/context_menu/audio_input_selector.h"
+#include "gui/context_menu/launch_style.h"
+#include "gui/menu_item/colour.h"
+#include "gui/menu_item/unpatched_param.h"
+#include "gui/ui/keyboard/keyboard_screen.h"
+#include "gui/ui/load/load_instrument_preset_ui.h"
+#include "gui/ui/load/load_song_ui.h"
+#include "gui/ui/menus.h"
+#include "gui/ui/ui.h"
+#include "gui/ui_timer_manager.h"
+#include "gui/views/arranger_view.h"
+#include "gui/views/audio_clip_view.h"
+#include "gui/views/automation_instrument_clip_view.h"
+#include "gui/views/instrument_clip_view.h"
+#include "gui/views/session_view.h"
+#include "gui/views/view.h"
+#include "hid/button.h"
+#include "hid/buttons.h"
+#include "hid/display/display.h"
+#include "hid/led/indicator_leds.h"
+#include "hid/led/pad_leds.h"
+#include "io/debug/print.h"
+#include "memory/general_memory_allocator.h"
+#include "model/action/action_logger.h"
+#include "model/clip/instrument_clip.h"
+#include "model/consequence/consequence_performance_layout_change.h"
+#include "model/settings/runtime_feature_settings.h"
+#include "model/song/song.h"
+#include "playback/mode/arrangement.h"
+#include "playback/playback_handler.h"
+#include "processing/engines/audio_engine.h"
+#include "storage/storage_manager.h"
+#include "util/d_string.h"
+#include "util/functions.h"
+#include <new>
+
+extern "C" {
+#include "RZA1/uart/sio_char.h"
+#include "util/cfunctions.h"
+}
+
+using namespace deluge;
+using namespace gui;
+
+const char* PERFORM_DEFAULTS_XML = "PerformanceView.XML";
```

Comment:
these should be defines or constexpr, not const

Reply:
Will make this change thanks!

Reply:
Done